controls/bsi_app_4_4.yml

```diff
@@ -313,51 +313,128 @@ controls:
+      recommended for all nodes. The correct version and configuration of RHCOS is verified 
+      cryptographically with the desired state, that is managed by the Control Plane using MachineConfigs. 
+      Any manual change on managed files is overwritten to ensure the desired state. Therefore, the 
+      control is mostly inheretly met when using CoreOS for all nodes.
+      
+      Section 1: OpenShift uses an internal Certificate Authority (CA). The nodes (kubelet to API server
+      and MachineConfig daemon to MachineConfi server) are communicating using node-specific certificates,
+      signed by this CA. Correct permissions of relevant files and secure TLS configuration are verified
+      using the referenced rules.
+
+      Section 2: Using the Red Hat File Integrity Operator, all files on the RHCOS nodes can be
+      cryptographically checked for integrity using Advanced Intrusion Detection Environment (AIDE).
+    status: automated
+    rules:
+      # Section 1 (worker / kubelet)
+      - file_groupowner_kubelet_conf
+      - file_groupowner_worker_ca
+      - file_groupowner_worker_kubeconfig
+      - file_groupowner_worker_service
+      - file_owner_kubelet
+      - file_owner_kubelet_conf
+      - file_owner_worker_ca
+      - file_owner_worker_kubeconfig
+      - file_owner_worker_service
+      - file_permissions_kubelet
+      - file_permissions_kubelet_conf
+      - file_permissions_worker_ca
+      - file_permissions_worker_kubeconfig
+      - file_permissions_worker_service
+      - kubelet_configure_client_ca
+      - kubelet_configure_tls_cert
+      - kubelet_configure_tls_cipher_suites
+      - kubelet_configure_tls_key
+      - kubelet_configure_tls_min_version
+      # Section 1 (API Server)
+      - api_server_client_ca
+      - api_server_kubelet_client_cert
+      - api_server_kubelet_client_key
+      - api_server_https_for_kubelet_conn
+      - api_server_tls_cert
+      - api_server_tls_cipher_suites
+      - api_server_tls_private_key
+      - api_server_tls_security_profile_not_old
+      - tls_version_check_apiserver
+      # Section 2
+      - cluster_version_operator_exists
+      - cluster_version_operator_verify_integrity
+      - file_integrity_exists
+      - file_integrity_notification_enabled
 
   - id: APP.4.4.A18
     title: Use of Micro-Segmentation
     levels:
     - elevated
     description: >-
-      Pods SHOULD ONLY be able to communicate with each other through the necessary network
-      ports, even within a Kubernetes namespace. There SHOULD be rules within the CNI that
-      disallow all but the necessary network connections within the Kubernetes namespace. These
+      (1) Pods SHOULD ONLY be able to communicate with each other through the necessary network
+      ports, even within a Kubernetes namespace. (2) There SHOULD be rules within the CNI that
+      disallow all but the necessary network connections within the Kubernetes namespace. (3) These

```

Comment:
not sure if a manual rule for section 3 would help or only create work.
What about cases, where someone creates "ANY-ANY" network policies and lovely selects all available pods?
Or is this more something we would expect to be solved in the application development/review cycle?

Reply:
I am actually surprised that noone has yet created such a rule yet.

However, I do not have a clear opinion yet, if the manual rule will add value or if this should be marked as "organizational measure". From a responsibility perspective, I would definately seet it at app dev/review.

Reply:
if i remember the discussions with the customers right, they opted for more rules (even if manual) instead of missing anything. There must be tailoring / review of the manual rules done at each customer anyway. Maybe a manual rule with the tenor "review netpols if they only include relevant connections and are using the criteria service name, labels, etc. pp. OR ensure there is a check in the app dev/review" this would make it transparent in the CO. 

On the other hand, you already mentioned this in the notes. But the notes are not visible in the CO outputs.

Reply:
I have created a manual rule for that.